pysb/core.py

```diff
@@ -988,6 +1006,8 @@ def __radd__(self, other):
             return NotImplemented
 
     def __mod__(self, other):
+        if self._tag is not None:
+            raise ValueError('Tag should be specified at the end of the complex')
```

Comment:
Is this actually a problem, or just bad style? If it's just a style issue, maybe a warning would be better.

Reply:
It's a problem in some scenarios.

`A() % A() @ tag % B()` generates this error. Since we can't nest complexes or tag a subset of a species, the tag would have to get applied to the whole species `A() % A() % B()`, which might not be what the modeler intended.

The error would also come up if trying to bind two tagged species, e.g. `(A() % A() @ tag1) % (B() % B() @ tag2)`. Since we can't have two tags attached to the single combined species `A() % A() % B() % B()`, we'd have to choose a tag by some precedence system (or error out).